src/app/clusters/administrator-commissioning-server/AdministratorCommissioningLogic.cpp

```diff
@@ -107,16 +107,32 @@ DataModel::ActionReturnStatus AdministratorCommissioningLogic::RevokeCommissioni
     MATTER_TRACE_SCOPE("RevokeCommissioning", "AdministratorCommissioning");
     ChipLogProgress(Zcl, "Received command to close commissioning window");
 
-    Server::GetInstance().GetFailSafeContext().ForceFailSafeTimerExpiry();
-
     if (!Server::GetInstance().GetCommissioningWindowManager().IsCommissioningWindowOpen())
     {
         ChipLogError(Zcl, "Commissioning window is currently not open");
         return ClusterStatusCode::ClusterSpecificFailure(StatusCode::kWindowNotOpen);
     }
 
-    Server::GetInstance().GetCommissioningWindowManager().CloseCommissioningWindow();
-    ChipLogProgress(Zcl, "Commissioning window is now closed");
+    auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
+
+    // No need to call ForceFailSafeTimerExpiry() if fail-safe is not armed
+    if (!failSafeContext.IsFailSafeArmed())
```

Comment:
So concretely why I think this is not the right approach: say we have the following sequence of events:

1. Commissioning window opened.
2. PASE session set up.
3. A packet arrives with two commands in it: ArmFailSafe to 0 and then RevokeCommissioning, which therefore get processed sync right after each other.

With the code in this PR, what happens?  As far as I can tell, it will have exactly the failure mode we have right now, where the failsafe will get disarmed and queue an event, then RevokeCommissioning will see it's disarmed and CloseCommissioningWindow, and the PASE session will be left dangling.